client/src/main/com/sinch/sdk/auth/adapters/VerificationApplicationAuthManager.java

```diff
@@ -0,0 +1,100 @@
+package com.sinch.sdk.auth.adapters;
+
+import com.sinch.sdk.core.exceptions.ApiAuthException;
+import com.sinch.sdk.core.exceptions.ApiException;
+import com.sinch.sdk.core.http.AuthManager;
+import com.sinch.sdk.core.utils.Pair;
+import com.sinch.sdk.core.utils.StringUtil;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collection;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+public class VerificationApplicationAuthManager implements AuthManager {
+
+  private static final String AUTH_KEYWORD = "Application";
+  private static final String XTIMESTAMP_HEADER = "x-timestamp";
+  private final String key;
+  private final byte[] secret;
+
+  public VerificationApplicationAuthManager(String key, String base64Secret) {
+    this.key = key;
+    this.secret = Base64.getDecoder().decode(base64Secret);
+  }
+
+  // FIXME: Verification OAS file claim it support "Basic" but miss the "Application" definition
+  public String getSchema() {
+    return "Basic";
+  }
+
+  @Override
+  public void resetToken() {
+    // no op
+  }
+
+  @Override
+  public Collection<Pair<String, String>> getAuthorizationHeaders(
+      String timestamp, String method, String httpContentType, String path, String body) {
+
+    String decodePath;
+    try {
+      decodePath = new URI(path).getPath();
+    } catch (URISyntaxException e) {
+      throw new ApiException(e);
+    }
+    // see
+    // https://developers.sinch.com/docs/verification/api-reference/authentication/signed-request/
+    String bodyMD5Hash = getBodyMD5Hash(body);
+    String stringToSign = getSignature(method, bodyMD5Hash, httpContentType, timestamp, decodePath);
+    String encoded = encode(stringToSign);
+    String key = this.key == null ? "" : this.key;
+
+    return Arrays.asList(
+        new Pair<>("Authorization", AUTH_KEYWORD + " " + key + ":" + encoded),
+        new Pair<>(XTIMESTAMP_HEADER, timestamp));
+  }
+
+  private String getBodyMD5Hash(String body) {
+    if (StringUtil.isEmpty(body)) {
+      return "";
+    }
+    try {
+      MessageDigest md = MessageDigest.getInstance("MD5");
```

Comment:
It used to be fine here: https://github.com/sinch/sinch-sdk-java/pull/15/files#diff-feb9f88279fca9ebde2e5c47af9da62694ad1d41b00f3e6f5f3672d6b51e3fa2R71

Reply:
Right (codeQL is only activated for push/PR onto main this is why there was no error before this PR)
About MD5, we will have to:
- ignore it because we have to follow the current algorithm used by Verification/Voice API
- send a message to Verification/Voice team to ensure they are aware about it